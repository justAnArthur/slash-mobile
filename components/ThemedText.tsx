import { StyleSheet, Text, type TextProps } from "react-native"

import { useOverrideThemeColor } from "@/lib/a11y/ThemeContext"

export type ThemedTextProps = TextProps & {
  lightColor?: string
  darkColor?: string
  type?:
    | "default"
    | "title"
    | "defaultSemiBold"
    | "subtitle"
    | "link"
    | "small"
    | "extraSmall"
}

export function ThemedText({
  style,
  lightColor,
  darkColor,
  type = "default",
  ...rest
}: ThemedTextProps) {
  const color = useOverrideThemeColor("text", {
    light: lightColor,
    dark: darkColor
  })

  return (
    <Text
      style={[
        { color },
        type === "default" ? styles.default : undefined,
        type === "title" ? styles.title : undefined,
        type === "defaultSemiBold" ? styles.defaultSemiBold : undefined,
        type === "subtitle" ? styles.subtitle : undefined,
        type === "link" ? styles.link : undefined,
        type === "small" ? styles.small : undefined,
        type === "extraSmall" ? styles.extraSmall : undefined,
        style
      ]}
      {...rest}
    />
  )
}

const styles = StyleSheet.create({
  default: {
    fontSize: 16,
    lineHeight: 24
  },
  defaultSemiBold: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: "600"
  },
  title: {
    fontSize: 32,
    fontWeight: "bold",
    lineHeight: 32
  },
  subtitle: {
    fontSize: 20,
    fontWeight: "bold"
  },
  link: {
    lineHeight: 30,
    fontSize: 16,
    color: "#0a7ea4"
  },
  extraSmall: {
    fontSize: 12
  },
  small: {
    fontSize: 14
  }
})
