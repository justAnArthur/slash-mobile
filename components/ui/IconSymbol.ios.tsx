<<<<<<< HEAD
import {
  SymbolView,
  type SymbolViewProps,
  type SymbolWeight
} from "expo-symbols"
import type { StyleProp, ViewStyle } from "react-native"

export function IconSymbol({
  name,
  size = 24,
  color,
  style,
  weight = "regular"
}: {
  name: SymbolViewProps["name"]
  size?: number
  color: string
  style?: StyleProp<ViewStyle>
  weight?: SymbolWeight
}) {
  return (
    <SymbolView
      weight={weight}
      tintColor={color}
      resizeMode="scaleAspectFit"
      name={name}
      style={[
        {
          width: size,
          height: size
        },
        style
      ]}
    />
  )
}
=======
import {
  SymbolView,
  type SymbolViewProps,
  type SymbolWeight
} from "expo-symbols"
import type { StyleProp, ViewStyle } from "react-native"

export function IconSymbol({
  name,
  size = 24,
  color,
  style,
  weight = "regular"
}: {
  name: SymbolViewProps["name"]
  size?: number
  color: string
  style?: StyleProp<ViewStyle>
  weight?: SymbolWeight
}) {
  return (
    <SymbolView
      weight={weight}
      tintColor={color}
      resizeMode="scaleAspectFit"
      name={name}
      style={[
        {
          width: size,
          height: size
        },
        style
      ]}
    />
  )
}
>>>>>>> 22703bd7fa6ddb9c5f3446763a1797c3b2ec69d8
