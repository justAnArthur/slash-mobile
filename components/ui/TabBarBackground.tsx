<<<<<<< HEAD
// This is a shim for web and Android where the tab bar is generally opaque.
export default undefined

export function useBottomTabOverflow() {
  return 0
}
=======
// This is a shim for web and Android where the tab bar is generally opaque.
export default undefined

export function useBottomTabOverflow() {
  return 0
}
>>>>>>> 22703bd7fa6ddb9c5f3446763a1797c3b2ec69d8
