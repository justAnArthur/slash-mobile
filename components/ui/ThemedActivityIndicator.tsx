import { useTheme } from "@/lib/a11y/ThemeContext"
import type React from "react"
import {
  ActivityIndicator,
  StyleSheet,
  View,
  type ViewProps
} from "react-native"

interface ThemedActivityIndicatorProps extends ViewProps {
  size?: "small" | "large"
}

export const ThemedActivityIndicator: React.FC<
  ThemedActivityIndicatorProps
> = ({ size = "small", style, ...otherProps }) => {
  const { theme } = useTheme()

  return (
    <View style={[styles.container, style]} {...otherProps}>
      <ActivityIndicator size={size} color={theme.foreground} />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
    height: "100%",
    flex: 1,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    padding: 10
  }
})

export const activityIndicatorStyles = styles
