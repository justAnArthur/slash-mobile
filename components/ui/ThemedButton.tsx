import { useTheme } from "@/lib/a11y/ThemeContext"
import React from "react"
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  type TouchableOpacityProps
} from "react-native"

interface ButtonProps extends TouchableOpacityProps {
  title: string
}

export const ThemedButton = ({ title, style, ...props }: ButtonProps) => {
  const { currentThemeMode } = useTheme()
  const isDarkMode = currentThemeMode === "dark"

  return (
    <TouchableOpacity
      {...props}
      style={[styles.button, isDarkMode && styles.buttonDark, style]}
      activeOpacity={0.7}
    >
      <Text style={[styles.text, isDarkMode && styles.textDark]}>{title}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    paddingHorizontal: 24,
    paddingVertical: 12,
    borderRadius: 32,
    backgroundColor: "rgba(40,40,40,.7)",
    boxShadow: "inset 2px 4px 16px 0 hsla(0,0%,97%,.06)",
    backdropFilter: "blur(50px)",
    fontSize: 14,
    // lineHeight: 1.4,
    fontWeight: "600",
    color: "hsla(0,0%,97%,.7)"
    // transition: "background .2s, color .2s"
  },
  buttonDark: {
    backgroundColor: "rgba(40,40,40,.7)"
  },
  text: {
    color: "hsla(0,0%,97%,.7)"
  },
  textDark: {
    color: "hsla(0,0%,97%,.7)"
  }
})

export const buttonStyles = styles
