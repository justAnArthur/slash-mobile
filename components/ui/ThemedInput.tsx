import { useTheme } from "@/lib/a11y/ThemeContext"
import React, { type PropsWithoutRef } from "react"
import { StyleSheet, TextInput, type TextInputProps } from "react-native"

export const ThemedInput = ({
  multiline,
  numberOfLines,
  style,
  ...props
}: PropsWithoutRef<TextInputProps>) => {
  const { theme } = useTheme()
  const styles = useStyles()

  return (
    <TextInput
      {...props}
      style={multiline ? [styles.textarea, style] : [styles.input, style]}
      placeholderTextColor={theme.mutedForeground}
      multiline={multiline}
      numberOfLines={multiline ? numberOfLines : 1}
    />
  )
}

function useStyles() {
  const { theme } = useTheme()

  return StyleSheet.create({
    input: {
      borderColor: theme.input,
      color: theme.foreground,
      width: "100%",
      borderRadius: 12,
      borderWidth: 2,
      fontSize: 14,
      paddingHorizontal: 18,
      paddingVertical: 12
    },
    textarea: {
      borderColor: theme.input,
      color: theme.foreground,
      width: "100%",
      borderRadius: 12,
      borderWidth: 2,
      fontSize: 14,
      lineHeight: 1.4,
      paddingLeft: 18,
      paddingRight: 18
    }
  })
}

export const useInputStyles = useStyles
