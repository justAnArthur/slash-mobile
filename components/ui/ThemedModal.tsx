import { useTheme } from "@/lib/a11y/ThemeContext"
import React from "react"
import {
  Modal,
  type ModalProps,
  StyleSheet,
  TouchableWithoutFeedback,
  View,
  type ViewProps
} from "react-native"

export type ThemedModalProps = ModalProps & {
  lightColor?: string
  darkColor?: string
  containerStyle?: ViewProps["style"]
}

export function ThemedModal({
  visible,
  onRequestClose,
  lightColor,
  darkColor,
  containerStyle,
  children,
  ...otherProps
}: ThemedModalProps) {
  const styles = useStyles()

  return (
    <Modal
      visible={visible}
      animationType="slide"
      presentationStyle="overFullScreen"
      transparent={true}
      {...otherProps}
    >
      <View style={styles.modalContainer}>
        <TouchableWithoutFeedback onPress={onRequestClose}>
          <View style={styles.background} />
        </TouchableWithoutFeedback>

        <View style={styles.modal}>{children}</View>
      </View>
    </Modal>
  )
}

function useStyles() {
  const { theme } = useTheme()

  return StyleSheet.create({
    button: {
      color: theme.cardForeground,
      padding: 24
    },
    confirmContainer: {
      display: "flex",
      gap: 8
    },
    modalContainer: {
      flex: 1,
      justifyContent: "flex-end"
    },
    background: {
      flex: 1,
      width: "100%"
    },
    modal: {
      borderColor: theme.border,
      width: "100%",
      borderRadius: 12,
      padding: 24,
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      borderWidth: 1.5,
      borderBottomWidth: 0,
      backgroundColor: theme.background
    }
  })
}
