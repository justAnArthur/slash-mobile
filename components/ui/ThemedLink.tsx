import { buttonStyles } from "@/components/ui/ThemedButton"
import { useTheme } from "@/lib/a11y/ThemeContext"
import { Link, type LinkProps } from "expo-router"

export const ThemedLink = ({ style, ...props }: LinkProps) => {
  const { currentThemeMode } = useTheme()
  const isDarkMode = currentThemeMode === "dark"

  return (
    <Link
      {...props}
      style={[
        buttonStyles.button,
        isDarkMode && buttonStyles.buttonDark,
        style
      ]}
    >
      {props.children}
    </Link>
  )
}
