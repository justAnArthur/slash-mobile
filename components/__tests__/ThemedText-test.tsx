<<<<<<< HEAD
import * as React from "react"
import renderer from "react-test-renderer"

import { ThemedText } from "../ui/ThemedText"

it("renders correctly", () => {
  const tree = renderer.create(<ThemedText>Snapshot test!</ThemedText>).toJSON()

  expect(tree).toMatchSnapshot()
})
=======
import * as React from "react"
import renderer from "react-test-renderer"

import { ThemedText } from "../ui/ThemedText"

it("renders correctly", () => {
  const tree = renderer.create(<ThemedText>Snapshot test!</ThemedText>).toJSON()

  expect(tree).toMatchSnapshot()
})
>>>>>>> 22703bd7fa6ddb9c5f3446763a1797c3b2ec69d8
