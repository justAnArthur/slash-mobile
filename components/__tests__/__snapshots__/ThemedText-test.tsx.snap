<<<<<<< HEAD
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
=======
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
>>>>>>> 22703bd7fa6ddb9c5f3446763a1797c3b2ec69d8
